integr8ly/integreatly-operator
# Port create alert manager monitoring resources

## JIRA

https://issues.jboss.org/browse/INTLY-4028

## Description
Porting of the `create_alertmanager.yml` tasks from monitoring stack configuration: https://github.com/integr8ly/installation/blob/master/roles/middleware_monitoring_config/tasks/main.yml#L8

## Verification

The steps below assume you are using CRC as your test environment.

- Authenticate with Openshift
- Run the following task in the Makefile: `make cluster/prepare/local`
- Run the operator locally: `operator-sdk up local --namespace "integreatly" --operator-flags "--products=monitoring"`
- Create instance of the installation CR:

```
apiVersion: integreatly.org/v1alpha1
kind: Installation
metadata:
  name: example-installation
spec:
  type: workshop
  namespacePrefix: integreatly-
  routingSubdomain: apps.com
  masterUrl: https://someurl.apps.com
  selfSignedCerts: true
  alertManager:
    alertManagerToEmail: test@email.com
    dmsWebHookURL: dms_test
    pdServiceKey: pds_test
    smtpAuthPassword: smtp_password_test
    smtpAuthUsername: smtp_username_test
    smtpSmartHost: smtp_smarthost_test
```
- Wait for reconcile of the monitoring product to complete
- In the `integreatly-middleware-monitoring` namespace, check to ensure that the `alertmanager-application-monitoring` secret includes the values from `alertmanager` spec, and that they are populated with the correct values. E.g.

```
global:
  resolve_timeout: 5m
  smtp_smarthost: smtp_smarthost_test
  smtp_from: noreply@alertmanager-route-integreatly-middleware-monitoring.apps-crc.testing
  smtp_auth_username: smtp_username_test
  smtp_auth_password: smtp_password_test
route:
  group_wait: 30s
  group_interval: 5m
  repeat_interval: 12h
  receiver: default
  routes:
  - match:
      severity: critical
    receiver: critical
  - match:
      alertname: DeadMansSwitch
    repeat_interval: 5m
    receiver: deadmansswitch
receivers:
- name: default
  email_configs:
  - send_resolved: true
    to: test@email.com
- name: critical
  pagerduty_configs:
  - service_key: pds_test
  email_configs:
  - send_resolved: true
    to: test@email.com
- name: deadmansswitch
  webhook_configs:
    - url: dms_test
inhibit_rules:
- source_match:
    alertname: 'JobRunningTimeExceeded'
    severity: 'critical'
  target_match:
    alertname: 'JobRunningTimeExceeded'
    severity: 'warning'
  equal: ['alertname', 'job', 'label_cronjob_name']
```

 - Check the logs of the `alertmanager- application-monitoring` pod of errors. You will see an error if the `smtp_hostname` value is not a valid URL and port number. Other then that no errors should be reported.
